lib/resources/postgres.rb

```diff
@@ -12,29 +13,111 @@ class Postgres < Inspec.resource(1)
     def initialize
       os = inspec.os
       if os.debian?
-        @service = 'postgresql'
-        @data_dir = '/var/lib/postgresql'
-        @version = inspec.command('ls /etc/postgresql/').stdout.chomp
-        @conf_dir = "/etc/postgresql/#{@version}/main"
+        #
+        # https://wiki.debian.org/PostgreSql
+        #
+        # Debian allows multiple versions of postgresql to be
+        # installed as well as multiple "clusters" to be configured.
+        #
+        version = version_from_dir("/etc/postgresql")
+        cluster = cluster_from_dir("/etc/postgresql/#{version}")
+        @conf_dir = "/etc/postgresql/#{version}/#{cluster}"
+        @data_dir = "/var/lib/postgresql/#{version}/#{cluster}"
       elsif os.redhat?
-        @service = 'postgresql'
-        @version = inspec.command('ls /var/lib/pgsql/').stdout.chomp
-        @data_dir = "/var/lib/pgsql/#{@version}/data"
+        #
+        # /var/lib/pgsql/data is the default data directory on RHEL6
+        # and RHEL7. However, PR #824 explicitly added version-based
+        # directories. Thus, we call #version_from_dir unless it looks
+        # like we are using unversioned directories.
+        #
+        # TODO(ssd): This has the potential to be noisy because of the
+        # warning in version_from_dir. We should determine which case
+        # is more common and only warn in the less common case.
+        #
+        version = if inspec.directory("/var/lib/pgsql/data").exist?
+                    warn "Found /var/lib/pgsql/data. Assuming postgresql install uses un-versioned directories."
```

Comment:
I agree.  I was gonna chat about this when more people are online because the logging style in this project in this project is a bit different than are usual ruby pattern.  Here we seem to create new per-class loggers with the option to pass them in to the constructor.  I like that pattern for some things, but for logging I think it clutters the method signatures and makes it hard to retrofit logging into existing classes/APIs.

In other projects we tend to prefer a global logging singleton object that we configure at the start of the application. So that adding logging to a class is as easy as:

```
require 'inspec/log'
Inspec::Log.warn("foo")
```

I noticed that inspec *does* have such a logging class already, but it isn't really used anywhere.